File: DesignPatternsExercises/java/ObserverDesignPattern/Solution/Observer.java
// public __________(7)__________ class Observer {
public abstract class Observer {
    protected BreakingNewsFeed breakingNewsFeed;

    public Observer(BreakingNewsFeed breakingNewsFeed) {
        this.breakingNewsFeed = breakingNewsFeed;
        // breakingNewsFeed.__________(8)__________;
        breakingNewsFeed.subscribe(this);
    }

    // public __________(9)__________ void __________(10)__________();
    public abstract void sendNotification();
}

File: DesignPatternsExercises/ruby/observer_design_pattern/solution/observer.rb
class Observer
  # def __________(4)__________(breaking_news_feed)
  def initialize(breaking_news_feed)
    # __________(5)__________ = breaking_news_feed
    @breaking_news_feed = breaking_news_feed
    # breaking_news_feed.subscribe(__________(6)__________)
    breaking_news_feed.subscribe(self)
  end

  def send_notification
    raise NotImplementedError, 'Subclasses must override this method'
  end
end
